lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectDatumWriter.java

```diff
@@ -139,6 +140,13 @@ else if (datum instanceof Short)
       datum = ((Short)datum).intValue();
     else if (datum instanceof Character)
         datum = (int)(char)(Character)datum;
+    else if (datum instanceof Map)
+      if (ReflectData.isMapWithNonStringKeysSchema(schema)) {
```

Comment:
Should this be `(datum instanceof Map) && isMapWithNonStringKeysSchema(schema)`?